src/WCSimLIGen.cc

```diff
@@ -0,0 +1,279 @@
+#include "WCSimLIGen.hh"
+#include "WCSimPrimaryGeneratorMessenger.hh"
+#include "WCSimPrimaryGeneratorAction.hh"
+
+#include "json.hpp"
+
+#include "G4ParticleGun.hh"
+#include "G4PhysicalConstants.hh"
+#include "G4OpticalPhoton.hh"
+#include "G4RandomDirection.hh"
+#include "G4SystemOfUnits.hh"
+
+using json = nlohmann::json;
+
+WCSimLIGen::WCSimLIGen(){
+
+    // Initialise
+    //myDetector = myDC;
+    //myMessenger = new WCSimLIGenMessenger(this);
+    this->Initialise();
+
+}
+
+WCSimLIGen::~WCSimLIGen(){
+
+    // things to delete
+    delete myLIGun;
```

Comment:
Missed this (less bad) leak last time
```suggestion
    delete myLIGun;
    if(hProfile) delete hProfile;
```